src/OpenLoco/Vehicles/VehicleBody.cpp

```diff
@@ -165,8 +165,8 @@ namespace OpenLoco::Vehicles
             }
             else
             {
-                ah = vehicle_arr_4F865C[frontBogie->var_2C >> 2];
-                if (((frontBogie->var_2C >> 3) == 12) || ((frontBogie->var_2C >> 3) == 13))
+                ah = vehicle_arr_4F865C[frontBogie->var_2C.track._data >> 2];
```

Comment:
this is interesting in that it takes the track id and the diagonal flag. I'm not sure if that is correct or not.